python/pyspark/daemon.py

```diff
@@ -160,6 +160,15 @@ def handle_sigterm(*args):
                 if pid == 0:
                     # in child process
                     listen_sock.close()
+
+                    # close stdin for child process, see SPARK-26175
+                    # but we cannot just close it, because it make fileno "0" unallocated
+                    # and cause later new created file descriptor use fileno "0"
+                    # so here I close fd 0 and reopen it on "/dev/null"
+                    dummy_input = open(os.devnull, 'r')
+                    os.dup2(dummy_input.fileno(), 0)
+                    dummy_input.close()
```

Comment:
```python
                    # It should close the standard input in the child process so that
                    # Python native function executions stay intact.
                    #
                    # Note that if we just close the standard input (file descriptor 0),
                    # the lowest file descriptor (file descriptor 0) will be allocated,
                    # later when other file descriptors should happen to open. 
                    #
                    # Therefore, here we redirects it to '/dev/null' by duplicating
                    # another file descriptor for '/dev/null' to the standard input (0).
                    # See SPARK-26175.
     
                    devnull = open(os.devnull, 'r')
                    os.dup2(devnull.fileno(), 0)
                    devnull.close()
```